numpyro/distributions/continuous.py

```diff
@@ -1680,38 +1680,46 @@ def infer_shapes(df, loc, scale_tril):
         return batch_shape, event_shape
 
 
-class MultivariateBeta(Distribution):
+class GaussianCopula(Distribution):
     arg_constraints = {
-        "concentration1": constraints.positive,
-        "concentration0": constraints.positive,
+        "covariance_matrix": constraints.positive_definite,
+        "precision_matrix": constraints.positive_definite,
+        "scale_tril": constraints.lower_cholesky,
     }
-    reparametrized_params = ["concentration1", "concentration0"]
+    support = constraints.real_vector
+    reparametrized_params = [
+        "covariance_matrix",
+        "precision_matrix",
+        "scale_tril",
+    ]
+
     support = constraints.unit_interval
 
     def __init__(
         self,
-        mv_normal,
-        concentration1,
-        concentration0,
+        transform_dist,
+        covariance_matrix=None,
+        precision_matrix=None,
+        scale_tril=None,
         validate_args=None,
     ):
-        self.concentration1, self.concentration0 = promote_shapes(
-            concentration1, concentration0
+        self.transform_dist = transform_dist
+        self.base_dist = MultivariateNormal(
+            covariance_matrix=covariance_matrix,
+            precision_matrix=precision_matrix,
+            scale_tril=scale_tril,
         )
+        self.normal = Normal(scale=jnp.diag(self.base_dist.covariance_matrix))
```

Comment:
`jnp.diag` only works for 1,2 dimensional arrays. I think you want `diagonal(..., -1, -2)` instead.